src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * The Ui class handles all of the Ui interaction with the users using its various static methods
+ */
+public class Ui {
+    private static String divider = "************************************************\n";
+
+    /**
+     * Introduces Duke to the users upon start up by printing an introduction.
+     */
+    public static void introduction() {
+        String intro = "Hello! I'm Duke\nWhat can i do for you?\n";
+        System.out.println(divider + intro + divider);
+    }
+
+    /**
+     * Prints out a done statement while updating the task string by swapping the cross with a tick.
+     * The doneTask argument must comply with the standard output string format of Duke tasks.
+     *
+     * @param doneTask the task that is to be marked done.
+     * @return returns the updated line.
+     */
+    public static String done(String doneTask) {
+        String updatedLine = doneTask.substring(0, 4) + "\u2713" + doneTask.substring(5);
+        System.out.println(divider + "Nice! I have marked this task as done:");
+        System.out.println(updatedLine + "\n" + divider);
+        return updatedLine;
+    }
+
+    /**
+     * Prints out the tasks in numerical order based on the ArrayList param.
+     * The lines argument must contain Strings that comply with the standard output string format of Duke tasks.
+     *
+     * @param lines List of tasks to be printed.
+     */
+    public static void listTasks(ArrayList<String> lines) {
+        System.out.println(divider);
+        System.out.println("Here are the tasks in your list!");
+        for (int i = 0; i < lines.size(); i++) {
+            int numbering = i + 1;
+            String task = lines.get(i);
+            System.out.println(numbering + "." + task);
+        }
+        System.out.println(divider);
+    }
+
+    /**
+<<<<<<< HEAD:src/main/java/duke/Ui.java
+     * Same as listTasks() except that the message printed out is slightly different. If the ArrayList supplied is
+     * empty, a "no matching tasks" message would be printed.
+     *
+     * @param lines The list of tasks to print out.
+     */
+    public static void listMatchingTasks(ArrayList<String> lines) {
+        System.out.println(divider);
+        if (lines.size() > 0) {
+            System.out.println("Here are the matching tasks in your list!");
+            for (int i = 0; i < lines.size(); i++) {
+                int numbering = i + 1;
+                String task = lines.get(i);
+                System.out.println(numbering + "." + task);
+            }
+        } else {
+            System.out.println("Unfortunately no tasks matches your keyword :(");
+        }
+        System.out.println(divider);
+    }
+
+
+    /**
+     * prints out a goodbye message when users exit Duke using the goodbye command
+     */
+    public static void bye() {
```

Comment:
I think that it can be renamed to exitProgram for a more intuitive understanding